lib/db/dao/topic_dao.dart

```diff
@@ -20,15 +20,18 @@ class TopicDao {
 
   Future<List<Topic>> getTopicsForCategoryId(String categoryId,
       {Database db}) async {
+           print("hello the id was in this is...::$categoryId");
     db = db ?? await new AppDatabase().getDb();
     List<Map> maps = await db.query(
         '${Topic.table} t, ${CategoryTopic.table} c',
-        columns: [Topic.idCol, Topic.nameCol, Topic.imageCol, Topic.colorCol],
+        columns: [Topic.idCol, Topic.nameCol, Topic.imageCol],
```

Comment:
Do not remove any columns from the query. Whenever a entity object is returned, it should be fully populated - all its fields should be there.
Add all the missing fields